frameworks/autosklearn/exec.py

```diff
@@ -42,53 +49,82 @@ def run(dataset, config):
         log.warning("Performance metric %s not supported.", config.metric)
 
     # Set resources based on datasize
-    log.info("Running auto-sklearn with a maximum time of %ss on %s cores with %sMB, optimizing %s.",
-             config.max_runtime_seconds, config.cores, config.max_mem_size_mb, perf_metric)
+    log.info(
+        "Running %s for %s with a maximum time of %ss on %s cores with %sMB, optimizing %s.",
+        askl_string,
+        dataset_name,
+        config.max_runtime_seconds,
+        config.cores,
+        config.max_mem_size_mb,
+        perf_metric,
+    )
     log.info("Environment: %s", os.environ)
 
     X_train = dataset.train.X_enc
     y_train = dataset.train.y_enc
     predictors_type = dataset.predictors_type
     log.debug("predictors_type=%s", predictors_type)
-    # log.info("finite=%s", np.isfinite(X_train))
 
     training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
 
     n_jobs = config.framework_params.get('_n_jobs', config.cores)
     ml_memory_limit = config.framework_params.get('_ml_memory_limit', 'auto')
-    ensemble_memory_limit = config.framework_params.get('_ensemble_memory_limit', 'auto')
 
-    # when memory is large enough, we should have:
-    # (cores - 1) * ml_memory_limit_mb + ensemble_memory_limit_mb = config.max_mem_size_mb
+    constr_params = {}
+    fit_extra_params = {'dataset_name': dataset_name}
+
     total_memory_mb = utils.system_memory_mb().total
     if ml_memory_limit == 'auto':
-        ml_memory_limit = max(min(config.max_mem_size_mb,
-                                  math.ceil(total_memory_mb / n_jobs)),
-                              3072)  # 3072 is autosklearn defaults
-    if ensemble_memory_limit == 'auto':
-        ensemble_memory_limit = max(math.ceil(ml_memory_limit - (total_memory_mb - config.max_mem_size_mb)),
-                                    math.ceil(ml_memory_limit / 3),  # default proportions
-                                    1024)  # 1024 is autosklearn defaults
-    log.info("Using %sMB memory per ML job and %sMB for ensemble job on a total of %s jobs.", ml_memory_limit, ensemble_memory_limit, n_jobs)
+        ml_memory_limit = max(
+            min(
+                config.max_mem_size_mb / n_jobs,
```

Comment:
@sebhrusen I think we should probably have some configuration error if `config.max_mem_size_mb < total_memory_mb`?

@mfeurer considering you are able to define a dynamic default here (`max(memory_limit/n_jobs, 3072)`), I am curious if there is a reason for not making this the default in the package itself?

Reply:
> I think we should probably have some configuration error if config.max_mem_size_mb < total_memory_mb?

app currently emits a warning, not an error. Main reason is to allow users to run benchmarks locally using any constraint, without worrying too much about the memory constraint.
Could add a config flag to switch this warning to error by default.

Reply:
> considering you are able to define a dynamic default here (max(memory_limit/n_jobs, 3072)), I am curious if there is a reason for not making this the default in the package itself?

I only updated the code which was there. I don't remember who wrote the original code. 

Changing this would only work for the single machine setup of the challenge (and most users), so I'm not sure yet if this would be a net improvement. I'll keep thinking about this, but don't know if we can change this before the benchmark cutoff.

Reply:
Yes that's fine. I just thought it would be something to consider.

Reply:
> Main reason is to allow users to run benchmarks locally using any constraint, without worrying too much about the memory constraint.

Ah, I think that's valid. Some switch could be a nice safeguard, made a note (#214) for later.